src/components/Map/Map.tsx

```diff
@@ -0,0 +1,119 @@
+import UrlIcon from 'leaflet/dist/images/marker-icon.png';
+import RetinaUrlIcon from 'leaflet/dist/images/marker-icon-2x.png';
+import ShadowUrlIcon from 'leaflet/dist/images/marker-shadow.png';
+import { icon } from 'leaflet';
+import {
+  AttributionControl,
+  MapContainer,
+  Marker,
+  TileLayer,
+  useMapEvents,
+} from 'react-leaflet';
+import React from 'react';
+
+import classes from './Map.module.css';
+import 'leaflet/dist/leaflet.css';
+
+// Default is center of Norway
+const DefaultCenter: [number, number] = [64.888996, 12.8186054];
+const DefaultZoom = 4;
+
+// Default map layers from Kartverket
+const DefaultMapLayers = [
+  {
+    url: 'https://opencache.statkart.no/gatekeeper/gk/gk.open_gmaps?layers=europa_forenklet&zoom={z}&x={x}&y={y}',
+    attribution: 'Data © <a href="http://www.kartverket.no/">Kartverket</a>',
+  },
+  {
+    url: 'https://opencache.statkart.no/gatekeeper/gk/gk.open_gmaps?layers=norgeskart_bakgrunn2&zoom={z}&x={x}&y={y}',
+    attribution: 'Data © <a href="http://www.kartverket.no/">Kartverket</a>',
+  },
+];
+
+export interface MapProps {
+  readOnly: boolean;
+  layers?: {
+    url: string;
+    attribution?: string;
+    subdomains?: string[];
+  }[];
+  center?: [number, number];
+  zoom?: number;
+  marker?: [number, number];
+  footerText?: string;
+  mapClicked?: (lat: number, lon: number) => void;
+}
+
+export const Map = ({
+  readOnly,
+  layers,
+  center,
+  zoom,
+  marker,
+  footerText,
+  mapClicked,
+}: MapProps) => {
+  layers = layers ? layers : DefaultMapLayers;
+  center = center ? center : DefaultCenter;
+  zoom = zoom ? zoom : DefaultZoom;
+
+  return (
+    <>
+      <MapContainer
+        className={classes.map}
+        center={center}
+        zoom={zoom}
+        zoomControl={!readOnly}
+        dragging={!readOnly}
+        touchZoom={!readOnly}
+        doubleClickZoom={!readOnly}
+        scrollWheelZoom={!readOnly}
+        attributionControl={false}
+      >
+        {layers.map((layer, i) => (
+          <TileLayer
+            key={i}
+            url={layer.url}
+            attribution={layer.attribution}
+            subdomains={layer.subdomains ? layer.subdomains : []}
+          />
+        ))}
+        <AttributionControl prefix={false} />
+        <LocationMarker
+          location={marker}
+          handleClick={mapClicked}
+        />
+      </MapContainer>
+      {footerText && <div className={classes.footer}>{footerText}</div>}
```

Comment:
I think the footer here is a bit redundant.
As the footer is optional anyway wouldn't it be easier to just remove this from the component entirely?

Reply:
I agree! It's better to handle the text footer outside of the component.

Reply:
I have now removed footer from the map component